UnderSea.BLL/Services/ArmyService.cs

```diff
@@ -0,0 +1,229 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UnderSea.BLL.DTO;
+using UnderSea.BLL.ViewModels;
+using UnderSea.DAL.Context;
+using UnderSea.DAL.Models;
+using UnderSea.DAL.Models.Units;
+
+namespace UnderSea.BLL.Services
+{
+    public class ArmyService : IArmyService
+    {
+        private readonly UnderSeaDbContext db;
+        private readonly ILogger logger;
+
+        public ArmyService(UnderSeaDbContext db, ILogger<ArmyService> logger)
+        {
+            this.db = db;
+            this.logger = logger;
+        }
+
+        public async Task<List<SimpleUnitViewModel>> Attack(int attackeruserid, AttackDTO attack)
+        { // TODO: majd ha egy játékosnak több országa lesz majd, akk itt az attackeruserid-ből attacking country id-t kéne csinálni
+            var game = await db.Game
+                .Include(game => game.Attacks)
+                .SingleAsync();
+            var unitTypes = await db.UnitTypes.ToListAsync();            
+            var attackingUser = await db.Users
+                .Include(u=>u.Country)
+                .ThenInclude(country => country.DefendingArmy)
+                .ThenInclude(defendingArmy => defendingArmy.Units)
+                .ThenInclude(unit => unit.Type)
+                .Include(u => u.Country)
+                .ThenInclude(country => country.AttackingArmy)
+                .ThenInclude(attackingArmy => attackingArmy.Units)
+                .ThenInclude(unit => unit.Type)
+                .SingleAsync(u => u.Id == attackeruserid);
+            var defendingCountry = await db.Countries
+                .Include(country => country.User)
+                .SingleAsync(c => c.UserId == attack.DefenderUserId);
+            var defendingUser = defendingCountry.User;
+
+            var sentUnits = new List<Unit>();
+            var newUnitGroup = new UnitGroup();
+            db.UnitGroups.Add(newUnitGroup);
+            await db.SaveChangesAsync();
```

Comment:
a végén elég visszaírni a db-be

Reply:
Azért mentettünk itt is, mert a támadáshoz létrehoztunk egy új UnitGroupot, aminek az Id-jára a foreachben szükség van a Unitok idegen kulcsaként